bindings/nodejs/examples/evm/send-evm-transaction.ts

```diff
@@ -0,0 +1,171 @@
+
+// This example uses secrets in environment variables for simplicity which should not be done in production.
+require('dotenv').config({ path: '.env' });
+
+const RPC_ENDPOINT = 'https://json-rpc.evm.testnet.shimmer.network';
+const RECIPIENT_ACCOUNT_ADDRESS = '0x2fF33407c26E36c32cA50A4e63ce661b2eeED3dd';
+const CHAIN_ID = 1072;
+const COIN_TYPE = 60;
+
+// fUSDC address. Tokens are available through: https://deepr.finance/faucet
+const TOKEN_CONTRACT_ADDRESS = '0x01ee95C34AeCAE1948aB618e467A6806b25fe7e4';
+const AMOUNT = 1e6; //fUSDC has 6 decimals
+
+const TX_OPTIONS = {
+    common: Common.custom({
+        chainId: CHAIN_ID,
+    }),
+    freeze: false,
+};
+
+async function run(): Promise<void> {
+    const provider = new Web3(RPC_ENDPOINT);
+    try {
+        if (!process.env.MNEMONIC) {
+            throw new Error('.env MNEMONIC is undefined, see .env.example');
+        }
+        const mnemonicSecretManager = {
+            mnemonic: process.env.MNEMONIC,
+        };
+
+        const secretManager = new SecretManager(mnemonicSecretManager);
+
+        const addresses = await secretManager.generateEvmAddresses({
+            coinType: COIN_TYPE,
+            accountIndex: 0,
+        });
+        const senderAddress = addresses[0];
+
+        // 1. Create unsigned transaction data
+        const txData = await createTxData(provider, senderAddress);
+        const transaction = Transaction.fromTxData(txData, TX_OPTIONS);
+
+        // 2. Create messageToSign by external signer
+        const message = transaction.getMessageToSign(false);
+        const serializedMessage = Buffer.from(RLP.encode(message));
+        const messageToSign = '0x' + serializedMessage.toString('hex');
+
+        // 3. Sign message with external signer
+        const bip44Path = {
+            coinType: 60,
+            account: 0,
+            change: 0,
+            addressIndex: 0,
+        };
+        const { signature } = await secretManager.signSecp256k1Ecdsa(
+            messageToSign,
+            bip44Path,
+        );
+
+        // 4. Make Secp256k1Ecdsa an Eip155Compatible Signature
+        const ecdsaSignature = fromRpcSig(signature);
+        ecdsaSignature.v = convertsVToEip155Compatible(
+            ecdsaSignature.v,
+            CHAIN_ID,
+        );
+
+        // 5. Sign Transaction
+        const signedTransaction = createSignedTransaction(
+            transaction,
+            ecdsaSignature,
+        );
+
+        // 7. Send signed transaction
+        const hexSignedTransaction =
+            getHexEncodedTransaction(signedTransaction);
+        const sentTransaction = await provider.eth.sendSignedTransaction(
+            hexSignedTransaction,
+        );
+        console.log('sent Transaction', sentTransaction);
+
+        // Testing: check sender address matches
+        strictEqual(
+            senderAddress,
+            signedTransaction.getSenderAddress().toString(),
+            'Mismatch in addresses',
+        );
```

Comment:
Shouldn't this be done before sending the transaction?

Reply:
Yeah, that makes sense. I updated the code.